examples/ErrorHandling/HandleKeywordPolicyViolations.php

```diff
@@ -0,0 +1,263 @@
+     * @param int $customerId the client customer ID without hyphens
+     * @param int $adGroupId the ad group ID to add a keyword to
+     * @param string $keywordText the keyword text to add
+     */
+    public static function runExample(
+        GoogleAdsClient $googleAdsClient,
+        int $customerId,
+        int $adGroupId,
+        string $keywordText
+    ) {
+        // Configures the keyword text and match type settings.
+        $keywordInfo = new KeywordInfo([
+            'text' => new StringValue(['value' => $keywordText]),
+            'match_type' => KeywordMatchType::EXACT
+        ]);
+
+        // Constructs an ad group criterion using the keyword text info above.
+        $adGroupCriterion = new AdGroupCriterion([
+            'ad_group' => new StringValue(
+                ['value' => ResourceNames::forAdGroup($customerId, $adGroupId)]
+            ),
+            'status' => AdGroupCriterionStatus::ENABLED,
+            'keyword' => $keywordInfo
+        ]);
+
+        $adGroupCriterionOperation = new AdGroupCriterionOperation();
+        $adGroupCriterionOperation->setCreate($adGroupCriterion);
+        $adGroupCriterionServiceClient = $googleAdsClient->getAdGroupCriterionServiceClient();
+
+        try {
+            // Try sending a mutate request to add the keyword.
+            $response = $adGroupCriterionServiceClient->mutateAdGroupCriteria(
+                $customerId,
+                [$adGroupCriterionOperation]
+            );
+            printf(
+                "Added a keyword with resource name '%s'.%s",
+                $response->getResults()[0]->getResourceName(),
+                PHP_EOL
+            );
+            // You may hit this line when you run this example using the default keyword for the
+            // second time. This is because the keyword will already be exempted and the mutate
+            // request can thus be successful.
+            return;
+        } catch (GoogleAdsException $googleAdsException) {
+            $exemptPolicyViolationKeys = self::fetchExemptPolicyViolationKeys($googleAdsException);
+        }
+
+        // Try sending exemption requests for creating a keyword. However, if your keyword contains
+        // many policy violations, but not all of them are exemptible, the request will not be sent.
+        if (count($exemptPolicyViolationKeys)
+            === count($googleAdsException->getGoogleAdsFailure()->getErrors())) {
+            self::requestExemption(
+                $customerId,
+                $adGroupCriterionServiceClient,
+                $adGroupCriterionOperation,
+                $exemptPolicyViolationKeys
+            );
+        } else {
+            print "No exemption request is sent because your keyword contained some non-exemptible"
+                . " policy violations" . PHP_EOL;
+        }
+    }
+
+    /**
+     * Collects all policy violation key that can be exempted for sending exemption request later.
```

Comment:
Typos: "key" and "request" should be in plural form "keys" and "requests"

Reply:
Done.